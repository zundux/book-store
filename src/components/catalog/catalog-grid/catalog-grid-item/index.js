import CatalogGridItem from "./catalog-grid-item";

export default CatalogGridItem;