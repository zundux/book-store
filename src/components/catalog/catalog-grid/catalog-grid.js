import React from "react";
import * as R from "ramda";
import CatalogGridItem from "./catalog-grid-item";

export const CatalogProductList = ({ products, onAddToCart }) =>
  products.map((product, productIndex) => (
    <CatalogGridItem
      key={productIndex}
      product={product}
      onAddToCart={onAddToCart}
    />
  ));

export const EmptyCatalogLabel = () => (
  <div>
    <p>Nothing found</p>
  </div>
);

const CatalogGrid = ({ products, onAddToCart }) =>
  R.ifElse(
    R.isEmpty,
    () => (
      <EmptyCatalogLabel/>
    ),
    () => (
      <CatalogProductList {...{
        products,
        onAddToCart
      }}/>
    )
  )(products);

export default CatalogGrid;