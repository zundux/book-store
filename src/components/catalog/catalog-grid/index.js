import CatalogGrid from "./catalog-grid";

export default CatalogGrid;