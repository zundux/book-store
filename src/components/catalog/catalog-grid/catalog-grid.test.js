import React from "react";
import CatalogGrid from ".";
import { CatalogProductList, EmptyCatalogLabel } from "./catalog-grid";

describe("Catalog grid", () => {

  test("Catalog grid has no products", () => {
    const products = [];
    const onAddToCart = () => {
    };

    const component = shallow(
      <CatalogGrid {...{
        products,
        onAddToCart
      }}/>
    );
    matchSnapshot(component);
  });

  test("Catalog grid has one products", () => {
    const products = [
      {
        id: 1,
        title: "product 1"
      }
    ];
    const onAddToCart = () => {
    };

    const component = shallow(
      <CatalogGrid {...{
        products,
        onAddToCart
      }}/>
    );
    matchSnapshot(component);
  });

  test("Catalog grid has a couple of products", () => {
    const products = [
      {
        id: 1,
        title: "product 1"
      }, {
        id: 2,
        title: "product 2"
      }
    ];
    const onAddToCart = () => {
    };

    const component = shallow(
      <CatalogGrid {...{
        products,
        onAddToCart
      }}/>
    );
    matchSnapshot(component);
  });


  test("Empty catalog label", () => {
    const component = shallow(
      <EmptyCatalogLabel/>
    );
    matchSnapshot(component);
  });


  test("Catalog product list has a couple of products", () => {
    const products = [
      {
        id: 1,
        title: "product 1"
      }, {
        id: 2,
        title: "product 2"
      }
    ];
    const onAddToCart = () => {
    };

    const component = shallow(
      <CatalogProductList {...{
        products,
        onAddToCart
      }}/>
    );
    matchSnapshot(component);
  });
});