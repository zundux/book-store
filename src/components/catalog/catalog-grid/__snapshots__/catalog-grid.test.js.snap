// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog grid Catalog grid has a couple of products 1`] = `
<Component
  onAddToCart={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "title": "product 1",
      },
      Object {
        "id": 2,
        "title": "product 2",
      },
    ]
  }
/>
`;

exports[`Catalog grid Catalog grid has no products 1`] = `<Component />`;

exports[`Catalog grid Catalog grid has one products 1`] = `
<Component
  onAddToCart={[Function]}
  products={
    Array [
      Object {
        "id": 1,
        "title": "product 1",
      },
    ]
  }
/>
`;

exports[`Catalog grid Catalog product list has a couple of products 1`] = `
Array [
  <CatalogGridItem
    key="0"
    onAddToCart={[Function]}
    product={
        Object {
            "id": 1,
            "title": "product 1",
          }
    }
/>,
  <CatalogGridItem
    key="1"
    onAddToCart={[Function]}
    product={
        Object {
            "id": 2,
            "title": "product 2",
          }
    }
/>,
]
`;

exports[`Catalog grid Empty catalog label 1`] = `
<div>
  <p>
    Nothing found
  </p>
</div>
`;
